packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -617,12 +665,19 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     _openOrCloseInputConnectionIfNeeded();
     _startOrStopCursorTimerIfNeeded();
     _updateOrDisposeSelectionOverlayIfNeeded();
-    if (!_hasFocus) {
+    if (_hasFocus) {
+      // Listen for changing viewInsets, which indicates keyboard showing up.
```

Comment:
Couldn't something else do that, like rotating the screen, or a "virtual" safe area element (like the iOS bottom button) disappearing? I suppose we probably want this to happen in those cases too, however.

Reply:
Yes, this will be triggered by the events you list as well. I'd argue that you'd still want to ensure that a focused textfield stays visible when any of those happen.